ddl/ddl_api.go

```diff
@@ -1731,22 +1731,11 @@ func buildTableInfoWithStmt(ctx sessionctx.Context, s *ast.CreateTableStmt, dbCh
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	switch s.TemporaryKeyword {
-	case ast.TemporaryGlobal:
-		tbInfo.TempTableType = model.TempTableGlobal
-		if !ctx.GetSessionVars().EnableGlobalTemporaryTable {
-			return nil, errors.New("global temporary table is experimental and it is switched off by tidb_enable_global_temporary_table")
-		}
-		// "create global temporary table ... on commit preserve rows"
-		if !s.OnCommitDelete {
-			return nil, errors.Trace(errUnsupportedOnCommitPreserve)
-		}
-	case ast.TemporaryLocal:
-		// TODO: set "tbInfo.TempTableType = model.TempTableLocal" after local temporary table is supported.
-		tbInfo.TempTableType = model.TempTableNone
+	if err = setTemporaryType(ctx, tbInfo, s); err != nil {
+		return nil, errors.Trace(err)
+	}
+	if s.TemporaryKeyword == ast.TemporaryLocal {
```

Comment:
I don't understand. Why leave this judgement out of `setTemporaryType`?

Reply:
put it into the method would AppendWarning twice

Reply:
`CreateTable` calls `buildTableInfoWithLike` and `buildTableInfoWithStmt`. So there are 2 ways:
1. Call `setTemporaryType` only in `CreateTable`, which is impossible. Because `setTemporaryType` must be called in the middle of `buildTableInfoWithStmt`, unless you remove the dependence of `tbInfo.TempTableType` in `buildTableInfoWithStmt`.
2. Call `setTemporaryType` both in `buildTableInfoWithLike` and `buildTableInfoWithStmt`.